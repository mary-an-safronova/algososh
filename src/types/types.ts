export enum ElementStates {
    Default = "default",
    Changing = "changing",
    Modified = "modified",
}
  
export type CircleElement = {
    value: string | number;
    state: ElementStates;
}

export type ColumnElement = {
    value: number;
    state: ElementStates;
}

export enum Direction {
    Ascending = "ascending",
    Descending = "descending",
}

export enum SortingMethod {
    Selection = "Выбор",
    Bubble = "Пузырёк",
}