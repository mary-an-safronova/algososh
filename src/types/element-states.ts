export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export type CircleElement = {
  value: string | number;
  state: ElementStates;
};
