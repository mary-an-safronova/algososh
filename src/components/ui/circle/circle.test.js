import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/types";

describe('Circle', () => {
    it('should renders correctly Circle without letter', () => {
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with letters', () => {
        const tree = renderer.create(<Circle letter="abc" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with head', () => {
        const tree = renderer.create(<Circle head="head" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with react element in the head', () => {
        const tree = renderer.create(<Circle isSmall={true} extraClass="topCircle" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with tail', () => {
        const tree = renderer.create(<Circle tail="tail" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with react element in the tail', () => {
        const tree = renderer.create(<Circle isSmall={true} extraClass="bottomCircle" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with index', () => {
        const tree = renderer.create(<Circle index={1} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle with prop "isSmall"', () => {
        const tree = renderer.create(<Circle isSmall={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle in state "default"', () => {
        const tree = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle in state "changing"', () => {
        const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Circle in state "modified"', () => {
        const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});