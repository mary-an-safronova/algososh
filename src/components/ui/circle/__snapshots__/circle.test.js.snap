// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should renders correctly Circle in state "changing" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle in state "default" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle in state "modified" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle react element in the head 1`] = `
<div
  className="content topCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle react element in the tail 1`] = `
<div
  className="content bottomCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with prop "isSmall" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with react element in the head 1`] = `
<div
  className="content topCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with react element in the tail 1`] = `
<div
  className="content bottomCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should renders correctly Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle should renders correctly Circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
